### PR TITLE
Add notification handler & prepared the notification redirect.

### PR DESCRIPTION
**Pull request by Solomon04**
- Handle notifications when a user is in the app.
- Clear app badge when user views notification
- Prepare the notification redirect when a user clicks a notification

### CODE REVIEW

File: navigation/index.tsx [hunk 1]
```diff
@@ -37,6 +38,13 @@ export default function Navigation({colorScheme}: { colorScheme: ColorSchemeName
             })
             .catch(() => console.error('Storage error has occurred'));
     }, []);
+    
+    useEffect(() => {
+        // Handle the notification events
+        Notifications.addNotificationResponseReceivedListener(response => {
+
```
Comment (Solomon04) [inline navigation/index.tsx:45]: @haron68 we need a way to redirect the user to the notification screen when they tap a notification. I am not sure the best way to do it other than adding code in every individual screen to redirect the user. However that is inefficient so I want to let you give it a try before I do that.

Comment (haron68) [reply navigation/index.tsx]: So this code in the index isn't working?

Comment (haron68) [reply navigation/index.tsx]: I think to solve this we have to use deeplinks

Comment (Solomon04) [reply navigation/index.tsx]: No I just didn't know how to redirect the user to the notification page

**Pull request merged**